packages/communicator-email-ses/awscfg.ts

```diff
@@ -0,0 +1,105 @@
+ *  AWS bits:
+ * my_aws_config:
+ *  aws_region: us-east-2
+ *  aws_access_key_id: ${AWS_ACCESS_KEY_ID}
+ *  aws_secret_access_key: ${AWS_SECRET_ACCESS_KEY}
+ *
+ * Then, each communicator module will support a list of AWS configurations, for
+ *  example:
+ * aws_ses_configurations: my_aws_config
+ * 
+ * By providing this list, the communicator can validate configuration information
+ *  at app startup.
+ * By default, these will just use a configuration called 'aws_config'.
+ * 
+ * When invoking the component, if there is more than one choice of config, it can
+ *   be specified to the call.  The default will be the first in the list.
+ */
+
+import { Error as DBOSError } from "@dbos-inc/dbos-sdk";
+
+interface ConfigProvider
+{
+    getConfig<T>(key: string): T | undefined;
+    getConfig<T>(key: string, defaultValue: T): T;
+}
+
+export interface AWSServiceConfig
+{
+    name: string,
+    region: string,
+    credentials: {
+      accessKeyId: string,
+      secretAccessKey: string,
+    },
+    httpOptions?: {
+        timeout?: number,
+        connectTimeout?: number,
+    },
+    maxRetries?: number
+}
+
+export interface AWSCfgFileItem
+{
+    aws_region?: string,
+    aws_access_key_id?: string,
+    aws_secret_access_key?: string,
+}
+
+export function loadAWSConfigByName(ctx: ConfigProvider, cfgname: string): AWSServiceConfig {
+    const cfgstrs = ctx.getConfig<AWSCfgFileItem|undefined>(cfgname, undefined);
+    if (!cfgstrs) {
+        throw new DBOSError.DBOSConfigKeyTypeError(cfgname, 'AWSCfgFileItem', 'null');
+    }
+
+    if (!cfgstrs.aws_region || typeof(cfgstrs.aws_region) !== 'string') {
+        throw new DBOSError.DBOSConfigKeyTypeError(`${cfgname}.aws_region`, 'string', typeof(cfgstrs.aws_region));
+    }
+    if (!cfgstrs.aws_access_key_id || typeof(cfgstrs.aws_access_key_id) !== 'string') {
+        throw new DBOSError.DBOSConfigKeyTypeError(`${cfgname}.aws_access_key_id`, 'string', typeof(cfgstrs.aws_access_key_id));
+    }
+    if (!cfgstrs.aws_secret_access_key || typeof(cfgstrs.aws_secret_access_key) !== 'string') {
+        throw new DBOSError.DBOSConfigKeyTypeError(`${cfgname}.aws_secret_access_key`, 'string', typeof(cfgstrs.aws_secret_access_key));
+    }
+
+    return {name: cfgname, region: cfgstrs.aws_region, credentials: {accessKeyId: cfgstrs.aws_access_key_id, secretAccessKey: cfgstrs.aws_secret_access_key}};
+}
+
+export function loadAWSCongfigsByNames(ctx: ConfigProvider, cfgnames: string) {
+    const configs: AWSServiceConfig[] = [];
+    const cfgnamesarr = (cfgnames || 'aws_config').split(',');
+
+    for (const cfgname of cfgnamesarr) {
+        configs.push(loadAWSConfigByName(ctx, cfgname))
+    }
+
+    return configs;
+}
+
+export function getAWSConfigs(ctx: ConfigProvider, svccfgname?: string) : AWSServiceConfig[]
+{
+    if (svccfgname) {
+        return loadAWSCongfigsByNames(ctx, ctx.getConfig<string>(svccfgname, ''));
+    }
+    return [loadAWSConfigByName(ctx, 'aws_config')];
+}
```

Comment:
What's the use case for multiple config? When you have more than 1 AWS service and you want to pick part of the config from different places? Or when you have multiple SES config in the same account (which I think you eluded to last time)?

This looks quite complicated :-)

Reply:
Multiple AWS service access keys in the same account.  It can be simple (add 3 values to config file) or complex, it depends on whether the user's setup is simple (1 AWS key) or complex (more than 1).